service/interpreter/InternalChannel.go

```diff
@@ -23,6 +26,16 @@ func (i *InternalChannel) GetAllReceived() map[string][]*iwfidl.EncodedObject {
 	return i.receivedData
 }
 
+func (i *InternalChannel) GetInfos() map[string]iwfidl.ChannelInfo {
+	infos := map[string]iwfidl.ChannelInfo{}
```

Comment:
Since we know the size, can we not create a map with a preallocated size here?
```suggestion
	infos := make(map[string]iwfidl.ChannelInfo, len(i.receivedData))
```